src/device/mod.rs

```diff
@@ -1460,12 +1463,37 @@ impl Runtime {
             .map(|peers| peers.iter().any(|p| p.public_key == exit_node.public_key))
             .unwrap_or_default();
 
+        self.requested_state.postquantum_wg = None;
+
         if is_meshnet_exit_node {
             if let Some(dns) = &self.entities.dns.lock().await.resolver {
                 self.reconfigure_dns_peer(dns, &dns.get_default_dns_servers())
                     .await?;
             }
-        } else if exit_node.endpoint.is_none() {
+        } else if let Some(addr) = exit_node.endpoint {
+            if let Some(pq_conf) = &self.features.post_quantum_vpn {
+                telio_log_debug!("Initializing PQ hanshake");
+
+                let fetch_keys = Box::pin(wg::pq::fetch_keys(
```

Comment:
:thinking: 

Are there any retry mechanisms for PQ handshake?

Reply:
No :D, what do you think it should look like? a loop with the timeout?

Reply:
Generally I see two viable options:
* Simpler one: a constant-period retry, this is how WG itself solves this problem. E.g. every 5s attempt to do the PQ-handshake. If the handshake progresses somewhere (e.g. first static WG is done, but PQ handshake does not respond, or if PQ handshake responds, but the WG one, where preshared key is already set - does not) and fails in some later step -> just restart it from the beginning
* Slightly fancier one: Do the same as above, but instead of constant-period retry, use exponential backoff strategy.

Reply:
Is it OK for `connect_exit_node()` to have a potentially infinite loop? The thing is that we need to set the PQ things before the wireguard interface state is consolidated

Reply:
So the solution to the problem is clear right now. However, it's going to be done in a follow-up PR based on LLT-4657

Reply:
Also keep in mind, that we will not be able to block external libtelio methods until handshake is done. We will need to do that in somewhat async manner, but spinning of separate module/task like we discussed is one way to go about it :).